docs/sql-migration-guide-upgrade.md

```diff
@@ -23,6 +23,9 @@ license: |
 {:toc}
 
 ## Upgrading From Spark SQL 2.4 to 3.0
+  - Since Spark 3.0, hive is upgraded to 2.3.x, so it is required to update the Hive
```

Comment:
This seems fairly incomplete to a reader. What database? 
hive -> Hive. This also needs to clarify that the Hive client was updated

Reply:
Hive maintains the `SCHEMA_VERSION` in the `VERSION` table of the metastore db(this  database name will be configured by the user in the hive-site.xml), this SCHEMA_VERSION should be updated to at least 2.3.0.  

 I taught it's better capture in the migration guide as in my production environment thrift server failed to start after upgrade to spark 3.0. If the users are aware of this then I can cancel this ticket

Reply:
@sandeep-katta How to reproduce this issue?

Reply:
configure the metastoredb in the hive-site.xml as below
 <property>
    <name>javax.jdo.option.ConnectionURL</name>
    <value>jdbc:mysql://vm1:3306/**sparksql**?characterEncoding=UTF-8</value>
  </property>

under **sparksql** database there will be one table by name VERSION with the below values


mysql> select * from VERSION;
+--------+----------------+--------------------------------------+
| VER_ID | SCHEMA_VERSION | VERSION_COMMENT                |
+--------+----------------+--------------------------------------+
|      1       |              **1.3.0**    |   Set by MetaStore root@127.0.0.1 |
+--------+----------------+--------------------------------------+
1 row in set (0.00 sec)

if you upgrade the spark to 3.0 and restart thrift server it will not start. You need to update SCHEMA_VERSION to at least **2.3.0**

Reply:
Do you know why it's `1.3.0`?  Hive 1.3.0 has not been released yet.

Reply:
Anyway, could you try to start thriftserver with `spark.sql.hive.metastore.version=1.2`?
```sh
sbin/start-thriftserver.sh --conf spark.sql.hive.metastore.version=1.2 --conf spark.sql.hive.metastore.jars=maven
```

Reply:
Till 2.4 I think Spark uses some private fork of Hive which is 1.2.1, so we updated the SCHEMA_VERSION to 1.3.0 which > 1.2.1

Reply:
Could you try to start thriftserver with `spark.sql.hive.metastore.version=1.2`?
```sh
sbin/start-thriftserver.sh --conf spark.sql.hive.metastore.version=1.2 --conf spark.sql.hive.metastore.jars=maven
```

Reply:
I can start the thrift server even by disabling schema verification `hive.metastore.schema.verification`

But by doing this we may not be able to get the benefits of upgrading to latest Hive version.

If the users are aware of this then I can close this PR.

Reply:
There are two things here:
1. If you want to improve the performance of the Hive Metastore Server. The correct way is to upgrade your Hive Metastore Server to latest version. `SCHEMA_VERSION` should be updated by [Hive itself](https://github.com/apache/hive/blob/c57a59611fa168ee38c6ee0ee60b1d6c4994f9f8/metastore/scripts/upgrade/mysql/upgrade-1.2.0-to-1.3.0.mysql.sql).
2. Upgrade built-in Hive to 2.3.x still can get benefits if you Hive Metastore Server is 1.2.x, such as [SPARK-12014](https://issues.apache.org/jira/browse/SPARK-12014), [SPARK-27500](https://issues.apache.org/jira/browse/SPARK-27500) and [SPARK-26321](https://issues.apache.org/jira/browse/SPARK-26321).

Reply:
okay understood, this should be in the scope of Hive upgrade. Thank you @wangyum . I will close this PR as invalid